### PR TITLE
Handling a home page

### LINKED ISSUE #16: Handle a home page
There is no way (at least a tested one) to configure a home page to a Citation website. Of course, it's a pretty obvious need.

The starting point will simply be to configure a page with a blank or `/` slug

### PR DESCRIPTION
**Pull request by Swiip**
fix #16 

While working on adding a home page I stumble upon the mechanism of preparing contents in window for server side rendering and rehidrating was wrong. So, the PR is quite larger than expected:

- new unit tests
- new algo for prepared contents in window
- fix react-router version as the beta version changes often
- handle root page for home

### CODE REVIEW

File: citation-react-renderer/src/load.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	}).default;
+});
+
+test('should work with a single page', async t => {
+	const pages = [page1];
+	const result = await load(serverUrl, pages);
+	t.deepEqual(result, {[componentId1]: content1});
+});
+
+test('should work with a two pages', async t => {
```
Comment (Slashgear) [inline citation-react-renderer/src/load.test.js]: 'should work with two pages'

File: citation-react-renderer/src/prepare.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	t.deepEqual(result, {[componentId1]: content1});
+});
+
+test('should work with a single home page', t => {
+	const url = {url: '', pages: [page1]};
+	const result = prepareContents(url, contents);
+	t.deepEqual(result, {[componentId1]: content1});
+});
+
+test('should work with a home page and a sub page', t => {
```
Comment (Slashgear) [inline citation-react-renderer/src/prepare.test.js]: should work with a+n home page and a sub page

File: citation-react-renderer/src/load.js [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
 	for (const page of pages) {
 		const children = page.children ? page.children : [];
-		const url = `${context}/${page.slug}`;
-		result = {
-			...result,
-			[url]: await queries.queryComponentTree(serverUrl, page.component),
-			...await load(serverUrl, children, url)
-		};
+		const childrenContent = await load(serverUrl, children);
+		result = {...result, ...childrenContent};
```
Comment (Slashgear) [inline citation-react-renderer/src/load.js:8]: ❤️

File: citation-react-renderer/src/urls.test.js [hunk 4]
```diff
@@ -1,18 +1,34 @@
...
-test('getPaths should work with one page', t => {
+test('should work with one page', t => {
 	const pages = [{slug: '1', children: []}];
 	const result = getPaths(pages);
-	t.deepEqual(result, ['/1']);
+	t.deepEqual(result, [{url: '/1', pages}]);
 });
 
-test('getPaths should work with one page and a children', t => {
+test('should work with one page and a children', t => {
```
Comment (Slashgear) [inline citation-react-renderer/src/urls.test.js]: 'should work with one page and a child~~ren~~'

File: citation-react-renderer/src/urls.test.js [hunk 5]
```diff
@@ -1,18 +1,34 @@
...
 	];
 	const result = getPaths(pages);
-	t.deepEqual(result, ['/1', '/1/2']);
+	t.deepEqual(result, [
+		{url: '/1', pages: [pages[0]]},
+		{url: '/1/2', pages: [pages[0], pages[0].children[0]]}
+	]);
+});
+
+test('should work with a home and one page', t => {
```
Comment (Slashgear) [inline citation-react-renderer/src/urls.test.js]: should work with a+n home and one page

**Review by Slashgear — COMMENTED** [review]
Some typos

**Comment by Swiip** [comment]
Thanks, fixed

**Comment by Slashgear** [comment]
LGTM 👍

**Pull request merged**